File: javaWebApp/src/main/java/com/saeinwoojoo/java/webapp/SplashActivity.java
package com.saeinwoojoo.java.webapp;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;

/**
 * Shows the splash screen.
 *
 */
public class SplashActivity extends BaseActivity {

    private static final int PROGRESS_UPDATE_DURATION = 20;

    private int mProgressMax;
    private int mProgressCnt;
    private Handler mHandler;
    private Runnable mRunnable;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        mProgressMax = getResources().getInteger(R.integer.SPLASH_PROGRESS_MAX);
        setTitleProgressbarMax(mProgressMax);
    }

    @Override
    protected void onResume() {
        super.onResume();
        startProgress();
    }

    @Override
    protected void onPause() {
        super.onPause();
        stopProgress();
    }

    private void stopProgress() {
        if (null != mHandler && null != mRunnable) {
            mHandler.removeCallbacks(mRunnable);
            mRunnable = null;
            mHandler = null;
        }
    }

    private void startProgress() {
        if (null == mHandler)
            mHandler = new Handler(Looper.myLooper());

        if (null == mRunnable) {
            mRunnable = new Runnable() {
                @Override
                public void run() {
                    ++mProgressCnt;
                    setTitleProgress(mProgressCnt);
                    if (mProgressMax <= mProgressCnt) {
                        Intent intent = new Intent(SplashActivity.this, MainActivity.class);
                        startActivity(intent);
                        finish();
                        return;
                    }
                    mHandler.postDelayed(mRunnable, PROGRESS_UPDATE_DURATION);
                }
            };

            mHandler.postDelayed(mRunnable, PROGRESS_UPDATE_DURATION);
        }
    }
}

File: kotlinWebApp/src/main/java/com/saeinwoojoo/kotlin/webapp/SplashActivity.kt
package com.saeinwoojoo.kotlin.webapp

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper

/**
 * Shows the splash screen.
 *
 */
class SplashActivity : BaseActivity() {

    private var mProgressMax: Int = 0
    private var mProgressCnt: Int = 0
    private var mHandler: Handler? = null
    private var mRunnable: Runnable? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        mProgressMax = resources.getInteger(R.integer.SPLASH_PROGRESS_MAX)
        setTitleProgressbarMax(mProgressMax)
    }

    override fun onResume() {
        super.onResume()
        startProgress()
    }

    override fun onPause() {
        super.onPause()
        stopProgress()
    }

    private fun stopProgress() {
        if (null != mHandler && null != mRunnable) {
            mHandler?.removeCallbacks(mRunnable)
            mRunnable = null
            mHandler = null
        }
    }

    private fun startProgress() {
        if (null == mHandler)
            mHandler = Handler(Looper.myLooper())

        if (null == mRunnable) {
            mRunnable = Runnable {
                ++mProgressCnt
                setTitleProgress(mProgressCnt)
                if (mProgressMax <= mProgressCnt) {
                    val intent = Intent(this@SplashActivity, MainActivity::class.java)
                    startActivity(intent)
                    finish()
                    return@Runnable
                }
                mHandler?.postDelayed(mRunnable, PROGRESS_UPDATE_DURATION.toLong())
            }
        }

        mHandler?.postDelayed(mRunnable, PROGRESS_UPDATE_DURATION.toLong())
    }

    companion object {

        private const val PROGRESS_UPDATE_DURATION = 20
    }
}
